src/iosrv/drpc_client.c

```diff
@@ -158,6 +160,126 @@ notify_bio_error(int media_err_type, int tgt_id)
 	return rc;
 }
 
+static int
+notify_ras_event(Mgmt__RASEvent *evt)
+{
+	Mgmt__ClusterEventReq	req = MGMT__CLUSTER_EVENT_REQ__INIT;
+	uint8_t		       *reqb;
+	size_t			reqb_size;
+	Drpc__Call	       *dreq;
+	Drpc__Response	       *dresp;
+	int			rc;
+
+	if (evt == NULL) {
+		D_ERROR("invalid RAS event\n");
+		return -DER_INVAL;
+	}
+
+	req.event_case = MGMT__CLUSTER_EVENT_REQ__EVENT_RAS;
+	req.ras = evt;
+
+	reqb_size = mgmt__cluster_event_req__get_packed_size(&req);
+	D_ALLOC(reqb, reqb_size);
+	if (reqb == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	mgmt__cluster_event_req__pack(&req, reqb);
+	rc = drpc_call_create(dss_drpc_ctx, DRPC_MODULE_MGMT,
+			      DRPC_METHOD_MGMT_CLUSTER_EVENT, &dreq);
+	if (rc != 0) {
+		D_FREE(reqb);
+		goto out;
+	}
+
+	dreq->body.len = reqb_size;
+	dreq->body.data = reqb;
+
+	D_DEBUG(DB_MGMT, "sending RAS event\n");
+
+	rc = drpc_call(dss_drpc_ctx, R_SYNC, dreq, &dresp);
+	if (rc != 0)
+		goto out_dreq;
+	if (dresp->status != DRPC__STATUS__SUCCESS) {
+		D_ERROR("received erroneous dRPC response: %d\n",
+			dresp->status);
+		rc = -DER_IO;
+	}
+
+	drpc_response_free(dresp);
+out_dreq:
+	/* This also frees reqb via dreq->body.data. */
+	drpc_call_free(dreq);
+out:
+	return rc;
+}
+
+int
+ds_notify_pool_svc_update(uuid_t pool_uuid, d_rank_list_t *svc_ranks)
+{
+	Mgmt__RASEvent		 evt = MGMT__RASEVENT__INIT;
+	Mgmt__PoolSvcEventInfo	 info = MGMT__POOL_SVC_EVENT_INFO__INIT;
+	struct timeval		 tv;
+	struct tm		*tm;
+
+	if (dss_drpc_ctx == NULL) {
+		D_ERROR("DRPC not connected\n");
+		return -DER_UNINIT;
+	}
+
+	if (pool_uuid == NULL) {
+		D_ERROR("invalid pool UUID\n");
+		return -DER_INVAL;
+	}
+
+	D_ALLOC(info.pool_uuid, DAOS_UUID_STR_SIZE);
+	if (info.pool_uuid == NULL) {
+		D_ERROR("failed to allocate pool uuid\n");
+		return -DER_NOMEM;
+	}
+	uuid_unparse_lower(pool_uuid, info.pool_uuid);
+
+	evt.extended_info_case = MGMT__RASEVENT__EXTENDED_INFO_POOL_SVC_INFO;
+	evt.pool_svc_info = &info;
+	evt.id = "pool_svc_ranks_update";
+
+	(void)gettimeofday(&tv, 0);
+	tm = localtime(&tv.tv_sec);
+	if (tm) {
+		D_ALLOC(evt.timestamp, DAOS_RAS_EVENT_STR_MAX_LEN);
+		snprintf(evt.timestamp, DAOS_RAS_EVENT_STR_MAX_LEN-1,
+			 " ts: [%04d/%02d/%02d-%02d:%02d:%02d.%02ld]",
+			 tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday,
+			 tm->tm_hour, tm->tm_min, tm->tm_sec,
+			 (long int) tv.tv_usec / 10000);
+	}
+
+//  /*
+//   * Event severity.
+//   */
+//  uint32_t severity;
+//  /*
+//   * Human readable message describing event.
+//   */
+//  char *msg;
+//  /*
+//   * Event type.
+//   */
+//  uint32_t type;
+//  /*
+//   * (optional) DAOS rank involved in event.
+//   */
+//  uint32_t rank;
+//  /*
+//   * (optional) Hostname of node involved in event.
+//   */
```

Comment:
(style)  Block comments use a trailing */ on a separate line
(style)  do not use C99 // comments